src/global/reco/ReconstructedJets_factory.cc

```diff
@@ -1,49 +1,63 @@
 // SPDX-License-Identifier: LGPL-3.0-or-later
 // Copyright (C) 2023 Zhongling Ji, Derek Anderson
 
-#include "ReconstructedJets_factory.h"
-
+// standard c includes
 #include <memory>
-
 #include <JANA/JEvent.h>
-#include "services/log/Log_service.h"
+#include <services/log/Log_service.h>
+// event data model definitions
+#include <edm4eic/ReconstructedParticle.h>
+// factory-specific includes
+#include "ReconstructedJets_factory.h"
+
 
 
 namespace eicrecon {
 
     void ReconstructedJets_factory::Init() {
 
-        // This prefix will be used for parameters
-        std::string param_prefix = "reco:" + GetTag();
-
-        // Set input data tags properly
-        InitDataTags(param_prefix);
-
         // SpdlogMixin logger initialization, sets m_log
-        InitLogger(GetApplication(), param_prefix, "info");
+        InitLogger(GetApplication(), GetPrefix(), "info");
 
+        // initialize jet reconstruction algorithm
         m_jet_algo.init(m_log);
-    }
 
-    void ReconstructedJets_factory::ChangeRun(const std::shared_ptr<const JEvent> &event) {
+    }  // end 'Init()'
+
+
+
+    void ReconstructedJets_factory::BeginRun(const std::shared_ptr<const JEvent> &event) {
+
         // Nothing to do here
-    }
+
+    }  // end 'BeginRun(std::shared_ptr<JEvent&>)'
+
+
 
     void ReconstructedJets_factory::Process(const std::shared_ptr<const JEvent> &event) {
-        auto rc_particles = event->Get<edm4eic::ReconstructedParticle>("ReconstructedParticles");
 
+        // grab input collection
+        auto input = event->Get<edm4eic::ReconstructedParticle>(GetInputTags()[0]);
```

Comment:
Avoid conversion to `std::vector`
```suggestion
        auto input = static_cast<const edm4eic::ReconstructedParticleCollection*>(event->GetCollectionBase(GetInputTags()[0]));
```